### ISSUE TITLE
[bitnami/elasticsearch] in Dualstack IPv6 & SVC mismatch

### ISSUE THREAD

**Issue by qdrddr** [bitnami/charts#24531]
### Name and Version

bitnami/elasticsearch

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Install bitnami/elasticsearch chart in k8s with DualStack IP configuration. 

I have an IP DualStack k8s configuration. Interestingly, elasticsearch in DualStack config fallback to using IPv6 and testing service connectivity via IPv6, but by default, it created an IPv4-only service.

So I needed to modify the `zammad-elasticsearch` & `zammad-elasticsearch-master-hl` services by modifying `spec` to these:

```yaml
Spec:
  ipFamilies:
    - IPv4
    - IPv6
  ipFamilyPolicy: PreferDualStack
```

Also, to make this work, I needed to remove `livenessProbe` & `readinessProbe` for the elasticsearch StetfulSet:
```
/spec/template/spec/containers/0/readinessProbe
/spec/template/spec/containers/0/livenessProbe
```
I discovered this using the official zammad helm chart 10.3.4, which uses elastic search as a sub-chart and opened a [ticket for them](https://github.com/zammad/zammad-helm/issues/261#issuecomment-2005367059).

### What is the expected behavior?

In the Dualstack configuration, if services are created using IPv4, then pods should communicate via that IPv4. 
If there are two types of IP, then even if one is reachable, elastic search should be working.
Or create services with IPv4 & IPv6 by default for DualStack k8s.

### What do you see instead?

Pods are trying to communicate via IPv6 to services with IPv4 and obviously failing.

**Comment by qdrddr** [comment]
Correct. This can be cured simply if the service would have IPv6 in DualStack k8s configurations.

But please note that there is also another angle to it: the pod has a bug/unexpected behavior in the DualStack config it trying to reach out to a service via IPv6 only, so in combination of not having IPv6 on the service it results in an issue, though the root cause is the pod using IPv6 while I would expect it to use IPv4 since it exists and IPv6 is not. @juan131

**Comment by juan131** [comment]
Thanks @qdrddr 

I created an internal task to update our [chart template](https://github.com/bitnami/charts/tree/main/template/CHART_NAME) and release a minor version of every single chart in the Bitnami catalog adding support for customizing IP families.

**Comment by carrodher** [comment]
This issue has been inactive for a long time and is being closed as part of routine maintenance. Due to other higher-priority work, we weren’t able to investigate this further. If the issue is still relevant and you’re interested in contributing a fix, we’d be happy to review a pull request.

Thanks for your understanding.
